### ISSUE TITLE
openssl dgst -macopt hexkey not supported

### ISSUE THREAD

**Issue by gotthardp** [openssl/openssl#16200]
The command
```
openssl dgst -sha256 -mac HMAC -macopt hexkey:FFFF -binary -out hmac.out
```
works on OpenSSL 1.1.1, but fails on 3.0.0 beta2 with
```
C00102FC167F0000:error:03000093:digital envelope routines:default_fixup_args:command not supported::0:[action:2, state:4] name=hexkey, value=FFFF
```

**Comment by dcooper16** [comment]
I came across this issue a couple of years ago (https://github.com/drwetter/testssl.sh/pull/1377). While I don't see anything at the moment saying that the above shouldn't work, the man page for `dgst` does say that the `mac` command is preferred. For OpenSSL 3.0.0, the above command may be replaced with:
```
openssl mac -digest sha256 -macopt hexkey:FFFF -binary -out hmac.out HMAC
```

**Comment by gotthardp** [comment]
Yeah, the trouble is that `mac` has been introduced in 3.0.0; it doesn't work with 1.1.1. The `dgst` should work with both, which is ideal for a migration period.

**Comment by dcooper16** [comment]
> Yeah, the trouble is that mac has been introduced in 3.0.0; it doesn't work with 1.1.1. The dgst should work with both, which is ideal for a migration period.

Understood. This is an issue we deal with a lot in testssl.sh, since it needs to work with many different versions of OpenSSL and LibreSSL. In many places, such as for computing an HMAC, we check what version of OpenSSL/LibreSSL is being used and then make the appropriate call based on that. In other places, we test the OpenSSL/LibreSSL being used to determine what capabilities it has, and then use that information to tailor the calls that testssl.sh makes.

**Comment by paulidale** [comment]
This was triaged at the last OTC call.  It's a bug, it needs to be fixed for 3.0.  It's a matter of when one of the fellows is able to prioritise it or someone else jumping in with a fix.

**Comment by paulidale** [comment]
It looks like legacy paths are being used to set this without the legacy key being set up.
